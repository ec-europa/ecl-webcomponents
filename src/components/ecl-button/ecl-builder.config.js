const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-button-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-button-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-button-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-button-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
