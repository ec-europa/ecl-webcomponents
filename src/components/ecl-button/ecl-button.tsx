import { Component, Prop, h, Element } from '@stencil/core';

@Component({
  tag: 'ecl-button',
  styleUrls: {
    ec: './build/styles/ecl-button-ec.css',
    eu: './build/styles/ecl-button-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclButton {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() type: string = 'primary';
  @Prop() theme: string = 'ec';

  componentDidRender() {
    if (this.el.getElementsByTagName('ecl-icon')[0] && this.el.querySelector('.ecl-icon')) {
      const slot = this.el.getElementsByTagName('ecl-icon')[0].getAttribute('slot');
      this.el.querySelector('.ecl-icon').classList.add('ecl-button__icon');
      if (slot) {
        this.el.querySelector('.ecl-icon').classList.add(`ecl-button__${slot.substring(0, 5) + '-' + slot.substring(5)}`);
      }
    }
  }

  getClass(): string {
    return [
      `ecl-button`,
      `ecl-button--${this.type}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <button class={this.getClass()}>
        <span class="ecl-button__container">
          <slot name="icon-before"></slot>
          <span class="ecl-button__label">
            <slot></slot>
          </span>
          <slot name="icon-after"></slot>
        </span>
      </button>
    )
  }
}
