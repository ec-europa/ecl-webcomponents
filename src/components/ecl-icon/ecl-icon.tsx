import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}