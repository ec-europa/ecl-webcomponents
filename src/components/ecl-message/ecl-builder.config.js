const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, './node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  scripts: [
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-message/message.js'),
      dest: path.resolve(outputFolder, 'scripts/ecl-message-vanilla.js'),
      options: {
        banner,
        moduleName: 'ECL',
        includePaths,
        sourceMap: false,
      },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-message-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-message-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-message-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-message-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
