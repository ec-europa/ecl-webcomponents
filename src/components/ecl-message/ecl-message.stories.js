const getArgs = () => ({
  variant: 'information',
  title: 'Information message',
  closeLabel: 'Close',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam accumsan semper lorem, ac mollis lacus tincidunt eu. Duis scelerisque diam eu tempus fringilla.',
  withUtils: true,
  eclScript: true,
});

const getArgTypes = () => ({
  variant: {
    type: { name: 'select' },
    options: ['information', 'success', 'warning', 'error'],
    table: {
      description: 'Message variant',
    },
  },
  title: {
    control: { name: 'string' },
    table: {
      description: 'Title of the message',
    },
  },
  description: {
    control: { name: 'string' },
    table: {
      description: 'Text of the message',
    },
  },
  closeLabel: {
    control: { name: 'string' },
    table: {
      description: 'Label of the close button',
    },
  },
});

export default {
  title: 'Components/message',
};

const Template = args =>
  `<ecl-message
    message-title="${args.title}"
    theme="${args.theme}"
    variant="${args.variant}"
    description="${args.description}"
    close-label="${args.closeLabel}"
    ecl-script
  >
    ${args.description}
  </ecl-message>`;

export const Message = Template.bind({});
Message.storyName = 'default';
Message.args = getArgs();
Message.argTypes = getArgTypes();
