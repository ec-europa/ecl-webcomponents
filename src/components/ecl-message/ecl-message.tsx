import { Component, h, Prop, getAssetPath, Element } from '@stencil/core';
declare const ECL: any;

@Component({
  tag: 'ecl-message',
  styleUrls: {
    ec: './build/styles/ecl-message-ec.css',
    eu: './build/styles/ecl-message-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})
export class EclMessage {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() variant: string = 'info';
  @Prop() styleClass: string;
  @Prop() messageTitle: string;
  @Prop() closeLabel: string;
  @Prop() eclScript: boolean = false;
  @Prop() withUtils: boolean = false;


  getClass(): string {
    return [
      `ecl-message`,
      `ecl-message--${this.getVariant()}`,
      this.styleClass
    ].join(' ');
  }

  getVariant() : string {
    return this.variant === 'information' ? 'info' : this.variant;
  }

  componentDidRender() {
    if (this.withUtils && !document.querySelector('#ecl-utils-styles')) {
      const style = document.createElement('style');
      style.id = 'ecl-utils-styles';
      style.innerHTML = `@import "./build/styles/ecl-utilities-${this.theme}.css"`;
      document.body.appendChild(style);
    }
    if (this.eclScript) {
      const src = getAssetPath('./build/scripts/ecl-message-vanilla.js');
      if (document.querySelector(`script[src="${src}"]`)) {
        document.querySelector(`script[src="${src}"]`).remove();
      }
      const script = document.createElement('script');
      script.src = src;
      script.onload = () => {
        const message = new ECL.Message(this.el.firstElementChild);
        message.init();
      };
      document.body.appendChild(script);
    }
  }

  render() {
    return (
      <div
        class={this.getClass()}
        data-ecl-message role="alert"
      >
        <ecl-icon 
          icon={this.variant}
          size={this.theme === 'ec' ? 'l' : 'm'}
          style-class={`ecl-message__icon sc-ecl-message-${this.theme}`}
        >
        </ecl-icon>
        <div class="ecl-message__content">
          <ecl-button
            type="ghost"
            style-class={`ecl-message__close sc-ecl-message-${this.theme}`}
            data-ecl-message-close
          >
            {this.closeLabel}
            <ecl-icon
              slot="icon-after"
              icon="close-filled"
              theme={this.theme}
              style-class="ecl-button__icon ecl-button__icon--after" 
            ></ecl-icon>
          </ecl-button>
          { this.messageTitle ?
            <div class="ecl-message__title">{this.messageTitle}</div>
           : '' }
          <div class="ecl-message__description">
            <slot></slot>
          </div>
        </div>
      </div>
    );
  }
}
