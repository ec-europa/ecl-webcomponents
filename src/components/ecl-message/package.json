{
  "name": "@ecl/web-components-message",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.0.0",
  "description": "ECL Message",
  "main": "ecl-message.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build:scripts": "cross-env ecl-builder scripts",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.7.0",
    "@ecl/vanilla-component-icon": "3.7.0",
    "@ecl/resources-ec-icons": "3.7.0",
    "@ecl/resources-eu-icons": "3.7.0",
    "@ecl/vanilla-component-message": "3.7.0"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}