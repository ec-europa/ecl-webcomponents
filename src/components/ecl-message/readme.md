# ecl-message



<!-- Auto Generated Below -->


## Properties

| Property       | Attribute       | Description | Type      | Default     |
| -------------- | --------------- | ----------- | --------- | ----------- |
| `closeLabel`   | `close-label`   |             | `string`  | `undefined` |
| `eclScript`    | `ecl-script`    |             | `boolean` | `false`     |
| `messageTitle` | `message-title` |             | `string`  | `undefined` |
| `styleClass`   | `style-class`   |             | `string`  | `undefined` |
| `theme`        | `theme`         |             | `string`  | `'ec'`      |
| `variant`      | `variant`       |             | `string`  | `'info'`    |
| `withUtils`    | `with-utils`    |             | `boolean` | `false`     |


## Dependencies

### Depends on

- [ecl-icon](../ecl-icon)
- [ecl-button](../ecl-button)

### Graph
```mermaid
graph TD;
  ecl-message --> ecl-icon
  ecl-message --> ecl-button
  style ecl-message fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
