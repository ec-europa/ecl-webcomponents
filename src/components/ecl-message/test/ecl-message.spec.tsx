import { newSpecPage } from '@stencil/core/testing';
import { EclMessage } from '../ecl-message';
import { EclButton } from '../../ecl-button/ecl-button';
import { EclIcon } from '../../ecl-icon/ecl-icon';

describe('ecl-message', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclMessage, EclButton, EclIcon],
      html: `<ecl-message
                message-title="Title for a message"
                theme="ec"
                variant="information"
                close-label="Close"
                ecl-script
              >
                Lorem ipsum dolor sit amet
              </ecl-message>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
