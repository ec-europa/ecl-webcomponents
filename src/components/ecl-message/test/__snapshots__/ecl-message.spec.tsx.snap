// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-message renders 1`] = `
<ecl-message close-label="Close" ecl-script="" message-title="Title for a message" theme="ec" variant="information" with-utils="">
  <!---->
  <div class="ecl-message ecl-message--info" data-ecl-message role="alert">
    <ecl-icon style-class="ecl-message__icon sc-ecl-message-ec">
      <svg class="ecl-icon ecl-icon--l ecl-message__icon sc-ecl-message-ec">
        <use xlink:href="/build/images/ec/icons.svg#information"></use>
      </svg>
    </ecl-icon>
    <div class="ecl-message__content">
      <ecl-button data-ecl-message-close style-class="ecl-message__close sc-ecl-message-ec">
        <!---->
        <button class="ecl-button ecl-button--ghost ecl-message__close sc-ecl-message-ec" data-ecl-message-close type="button">
          <span class="ecl-button__container">
            <span class="ecl-button__label">
              Close
            </span>
            <ecl-icon slot="icon-after" style-class="ecl-button__icon ecl-button__icon--after">
              <svg class="ecl-button__icon ecl-button__icon--after ecl-icon ecl-icon--xs sc-ecl-button-ec">
                <use xlink:href="/build/images/ec/icons.svg#close-filled"></use>
              </svg>
            </ecl-icon>
          </span>
        </button>
      </ecl-button>
      <div class="ecl-message__title">
        Title for a message
      </div>
      <div class="ecl-message__description">
        Lorem ipsum dolor sit amet
      </div>
    </div>
  </div>
</ecl-message>
`;
